core/types/calls.cc

```diff
@@ -4070,6 +4070,25 @@ class Array_compact : public IntrinsicMethod {
     }
 } Array_compact;
 
+class Hash_compact : public IntrinsicMethod {
+public:
+    void apply(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res) const override {
+        TypePtr element;
+
+        auto *ap = cast_type<AppliedType>(args.thisType);
+        ENFORCE(ap->klass == Symbols::Hash() || ap->klass.data(gs)->derivesFrom(gs, Symbols::Hash()));
+        ENFORCE(ap->targs.size() == 3);
```

Comment:
This program will break this ENFORCE.

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20strict%0A%0Aclass%20MyHash%20%3C%20Hash%0A%20%20extend%20T%3A%3AGeneric%0A%0A%20%20K%20%3D%20type_member%28%3Aout%29%0A%20%20V%20%3D%20type_member%28%3Aout%29%0A%20%20Elem%20%3D%20type_member%28%3Aout%29%0A%0A%20%20Another%20%3D%20type_member%28%3Aout%29%0Aend%0A%0AMyHash.new.compact)

```ruby
# typed: strict

class MyHash < Hash
  extend T::Generic

  K = type_member(:out)
  V = type_member(:out)
  Elem = type_member(:out)

  Another = type_member(:out)
end

MyHash.new.compact
```

Reply:
Make sense, I'll change the condition to be `ap->targs.size() >= 3` and will add this test case